File: DoItTest/ClientApp/src/domain/tests/items/answerOption/textFildAnswerOption.ts
import { TestItemType } from "../testItemType"
import { AnswerOption as AnswerOption } from "./answerOption"

export class TextFildAnswerOption extends AnswerOption {
    constructor(
        public id: string,
        public testItemId: string,
        public type: TestItemType,
        public answer: string
    ) {
        super(id, testItemId, type)
    }
}

export function mapTextFildAnswerOption(value: any): TextFildAnswerOption {
    return new TextFildAnswerOption(value.id, value.testItemId, value.type, value.answer)
}

File: DoItTest.Domain/Tests/TestItems/AnswerOptions/TextFildAnswerOption.cs
namespace DoItTest.Domain.Tests.TestItems.AnswerOptions
{
    public class TextFildAnswerOption : AnswerOption
    {
        public String? Answer { get; }

        public TextFildAnswerOption(Guid id, Guid testItemId, TestItemType type, String? answer) : base(id, testItemId, type)
        {
            Answer = answer;
        }
    }
}
